src/ml_tooling/baseclass.py

```diff
@@ -488,6 +490,72 @@ def gridsearch(
 
         return self.result[0].model, self.result
 
+    def bayesian_optim(
```

Comment:
Can we call the method `bayesian_search`? That way it's consistent with `gridsearch` and `random_search`

Reply:
Makes sense. Done!